airflow/models.py

```diff
@@ -830,7 +830,7 @@ def are_dependencies_met(
                 TI.task_id == task.task_id,
                 TI.execution_date ==
                     self.task.dag.previous_schedule(self.execution_date),
-                TI.state == State.SUCCESS,
+                TI.state.in_(State.SUCCESS, State.SKIPPED),
```

Comment:
Unfortunately, it isn't! This was my first inclination as well. I added a new example dag as part of my commit to exercise  depends_on_past=True with a branch operator that alternates taking one of 2 branches. This change alone did not help, though now I suspect another possible cause.